### DISCUSSION TITLE
How to use alias with macro call in alias

### DISCUSSION THREAD

**Discussion by jd-solanki** [xonsh/xonsh#4808] [Q&A] [answered]
Hi, I have an alias to run bash commands:
```python
aliases['bx'] = 'bash -c!'
```

I also another another alias that uses above alias for killing process:
```python
aliases['kp'] = 'bx sudo kill -9 $(sudo lsof -t -i:$arg0)'
```

But I am unable to use `bx` alias in `kp` alias.

Thanks :)

**Comment by anki-code** [comment, marked as answer]
First of all you can try [xontrib-sh](https://github.com/anki-code/xontrib-sh) for day by day pasting the bash commands.

Second you can use [ExecAlias](https://xon.sh/tutorial.html#aliases) for your case: `aliases['kp'] = """bash -c @('sudo echo ' + $(echo @($arg0)).strip())"""`

To trace how it works:
```python
$XONSH_TRACE_SUBPROC = True
aliases['kp'] = """bash -c @('sudo echo ' + $(echo @($arg0)).strip())"""
kp 111 222 333
# TRACE SUBPROC: (['kp', '111', '222', '333'],), captured=hiddenobject
# TRACE SUBPROC: (['echo', '111'],), captured=stdout
# TRACE SUBPROC: (['bash', '-c', 'sudo echo 111'],), captured=hiddenobject
# 111
```

Finally you can't use [subprocess macro calls](https://xon.sh/tutorial_macros.html#subprocess-macros) as alias. This looks like feature.
